Set up eslint

catrope opened the pull request:
Set up eslint globally, rather than locally in each package:
- Move eslint and related devDependencies to the top level
- Remove package-level lint commands, instead run one top-level lint
  command
- Create a top-level .eslintrc.json with common rules, which each of the
  package-specific .eslintrc.json files inherits from
This is also how Vue manages linting in their monorepo, and one
advantage is that top-level files (like package.json) are also linted.

egardner reviewed (changes requested):
This needs to be re-worked to get rid of the Yarn stuff.

I didn't know about the "recommendations" thing though – perhaps we could use that to recommend Volar and similar useful tools for Vue development.

catrope commented:
I've made major updates to this PR. The VSCode recommendations stuff is gone, we don't need that anymore now that we've moved away from Yarn. I've also changed it from a local eslint setup in just the vue-components package to a global setup for the entire repo, and updated existing code to fix eslint errors/warnings.
